### PR TITLE
Auth API: add zone lookup by /zones?zone=example.org.

### PR DESCRIPTION
**Pull request by zeha**
### Short description
Implements #5736.

Someone else please type in docs.

### Checklist
I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [ ] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [X] added or modified unit test(s)

### CODE REVIEW

**Comment by pieterlexis** [comment]
TODO:

- [ ] Swagger update

File: pdns/ws-auth.cc [hunk 1]
```diff
@@ -1390,7 +1390,20 @@ static void apiServerZones(HttpRequest* req, HttpResponse* resp) {
     throw HttpMethodNotAllowedException();
 
   vector<DomainInfo> domains;
-  B.getAllDomains(&domains, true); // incl. disabled
+
+  auto zone_it = req->getvars.find("zone");
+  if (zone_it != req->getvars.end()) {
```
Comment (chbruyand) [inline pdns/ws-auth.cc]: I think you can check `req->getvars.count("zone") != 0` and then use `req->getvars["zone"]`.

**Comment by zeha** [comment]
> calling /zones/example.org would make more sense IMO

Well yes, but then we're back in silly URL encoding land, where it's not clear how a zone called `foo/bar.example.org` would be encoded correctly by all possible clients.

File: pdns/ws-auth.cc (hunk 1 above)
Comment (zeha) [reply pdns/ws-auth.cc]: Yeah, but that sounded like doing the work twice?

Comment (chbruyand) [reply pdns/ws-auth.cc]: That's just that something like the following looks better IMHO :
```
if (req->getvars.count("zone") != 0) {
  auto zone = req->getvars["zone"];
}
```

Comment (zeha) [reply pdns/ws-auth.cc]: Pushed the style change.

**Comment by zeha** [comment]
@pieterlexis did you find some time to type in the swagger changes? <3

**Comment by zeha** [comment]
@pieterlexis thanks!

**Pull request merged**